app/lib/account/agent.dart

```diff
@@ -104,16 +158,31 @@ class AuthenticatedGithubAction implements AuthenticatedAgent {
   AuthenticatedGithubAction({
     required this.idToken,
     required this.payload,
-  });
+  }) {
+    _checkRepository(payload.repository);
+  }
 
   @override
   String? get email => null;
 }
 
+void _checkRepository(String repository) {
+  InvalidInputException.check(repository.trim().isNotEmpty,
+      'The JWT from GitHub must have a non-empty `repository`.');
+  final parts = repository.split('/');
+  InvalidInputException.check(
+      parts.length == 2, 'The JWT from GitHub must have a valid `repository`.');
+  InvalidInputException.check(parts.every((e) => e.isNotEmpty),
+      'The JWT from GitHub must have a valid `repository`.');
```

Comment:
This will produce a 400 error if they are not caught.

It's unclear who is responsible for ensuring that these properties are satisfied.

I think we have the following options:

**Option A:** The caller is responsible for ensuring that these properties are satisfied. If they are not the caller should have return 401 or 403, that's irrelevant to the discussion of `AuthenticatedGithubAction`, because as far as `AuthenticatedGithubAction` is concerned the caller should always check these properties are correct. If this is the behavior we want we should:
 * (A.1) Throw `ArgumentError`.
 * (A.2) Document the requirements on the input in the constructor for `AuthenticatedGithubAction`.

**Option B:** Whatever happen should never be able to happen. Then throw `AssertionError`. No need to check anything, or document it. But if this ever happens it's must be because some invariant is wrong and it implies a bug in our understanding of our own code.
 * (B.1) Throw `AssertionError`.

**Option C:** This function is part of the authentication logic and the security of the application depends on this logic. It seems a bit confusing that the constructor for an `AuthenticatedGithubAction` object contain elements critical to the authentication flow. If this is the case, then:
  * (C.1) Throw `AuthenticationException` which becomes a 401.

**Option D:** This function is part of the authorization logic, again this seems a bit confusing. Especially, since the constructor for `AuthenticatedGithubAction` doesn't have the context to know what the caller is trying to do (so how would it determine if the caller is authorized to do something). But if this was the case:
 * (D.1) Throw `AuthorizationException` which becomes a 403.

**Option E:** This function is actually parsing content from the JWT; thus, it might be appropriate to throw a `FormatException`.
 * (E.1) Throw `FormatException`.
 * (E.2) Document in the constructor that it throws a `FormatException` (all exceptions thrown should be documented, errors need not be)
 * (E.3) Catch the `FormatException` everywhere we call the constructor.

Throwing `InvalidInputException` causes a 400 error, I don't see how that makes sense. If this is due to invalid input, it is not invalid input on the part of the caller, but on the part of the JWT passed from Github.

I suspect that (A) or (B) or (E) is the correct choice.